cpp/bench/ann/src/cuvs/cuvs_ann_bench_param_parser.h

```diff
@@ -56,6 +56,24 @@ extern template class cuvs::bench::cuvs_cagra<int8_t, uint32_t>;
 #include "cuvs_mg_cagra_wrapper.h"
 #endif
 
+template <typename ParamT>
+void parse_dynamic_batching_params(const nlohmann::json& conf, ParamT& param)
+{
+  if (!conf.value("dynamic_batching", false)) { return; }
+  param.dynamic_batching = true;
+  if (conf.contains("dynamic_batching_max_batch_size")) {
+    param.dynamic_batching_max_batch_size = conf.at("dynamic_batching_max_batch_size");
+  }
+  if (conf.contains("dynamic_batching_soft_deadline_ms")) {
+    param.dynamic_batching_soft_deadline_ms = conf.at("dynamic_batching_soft_deadline_ms");
+  }
+  if (conf.contains("dynamic_batching_n_queues")) {
+    param.dynamic_batching_n_queues = conf.at("dynamic_batching_n_queues");
+  }
+  param.dynamic_batching_k =
+    uint32_t(uint32_t(conf.at("k")) * float(conf.value("refine_ratio", 1.0f)));
```

Comment:
That's because `k` is surprisingly not a search parameter in `ann_bench`. Instead, it's an argument of the `search` function. The config still provides this value for the algorithms, but they need custom logic to parse and store it (for access in `set_search_param` function which doesn't have it as an argument otherwise). In this case, the batcher needs to know the maximum number of neighbors it will need to return (to allocate the buffer).

Reply:
Good point! I've added a check that issues a warning and falls back to a non-batched behavior for the given request.